### DISCUSSION TITLE
ChoiceField in rest_framework.fields not support IntEnum

### DISCUSSION THREAD

**Discussion by bursonyang** [encode/django-rest-framework#8945] [General]
### Title
ChoiceField in rest_framework.fields not support IntEnum
### env
Python 3.10.1
djangorestframework==3.14.0

### question
I have a model defined like this:
```
class JobStatus(IntEnum):
    CREATE = 0
    DISPATCH = 1
class Job(models.Model):
    JOB_STATUS_CHOICES = [
            (JobStatus.CREATE, "CREATE"),
            (JobStatus.DISPATCH, "DISPATCH")
        ]
    status = models.IntegerField(choices=JOB_STATUS_CHOICES, default=JobStatus.CREATE)
```
When I post a request to save a job with status=1, the server give me a *\"0\" is not a valid choice.* error.
I reviewed the fields.py and found the 'to_internal_value' method:
```
def to_internal_value(self, data):
        if data == '' and self.allow_blank:
            return ''

        try:
            return self.choice_strings_to_values[str(data)]
        except KeyError:
            self.fail('invalid_choice', input=data)
``` 
Maybe the 'to_internal_value' couldn't handle the IntEnum type choice correctly. 
How do you think about it?

**Comment by auvipy** [comment]
does django already support it?

**Reply by bursonyang** [reply]
Yes, the`validate` in django field use "==" to check the post value

**Reply by auvipy** [reply]
then you are most welcome to contribute the compatibility fix
